Automattic/wp-calypso
# Ensure create-site redirect works in environments that use wpcom-user-bootstrap

Related to: #43569 and #43578 

The redirect logic in Gutenboarding's `OnboardingEdit` component checks for a `?new` query param, and if it exists, we're not already creating a site (and one hasn't been created), the user is logged in, and the user is able to access the latter steps of the signup flow, then we kick off creating a site.

Reported in #43569 and discussed elsewhere, in WordPress.com (non-development / testing environments) after logging in and being redirected to create a site, the user is directed to the plans step, instead of the site being created. After a bit of digging, this appears to be because of the `wpcom-user-bootstrap` setting in these production-like environments, that mean that the user is server-rendered and available at `window.currentUser`.

When Gutenboarding boots, if the `window.currentUser` object exists, [we continue on](https://github.com/automattic/wp-calypso/blob/5d71a9025c67a5f610b0d3023f26471fd4dcf1b8/client/landing/gutenboarding/index.tsx#L165) instead of waiting for the currentUser AJAX call to resolve. Because we're currently using the selector to get the `currentUser` in the `OnboardingEdit` component, in these production environments the currentUser seemed to be undefined at boot, even though the user is logged in.

This PR proposes checking for `window.currentUser` in the `getCurrentUser` resolver to handle both cases for checking for a current user.

#### Changes proposed in this Pull Request

* In the `getCurrentUser` resolver get the current user from `window.currentUser` as a first priority, then fallback to getting it via the API call to `/me`.

#### Testing instructions



* From a logged out user go to `/new` and complete signup until you get to the account creation step
* Click 'Log in'
* Log in to your account, and you should be redirected to the site being created

Test Production-like environments. 

Follow PCYsg-5YE-p2 to set `"wpcom-user-bootstrap": true`, and while logged in go through `/new` all the way up to the plans step. Then manually enter into the address bar `http://calypso.localhost:3000/new/create-site?new` and it should start creating a site and redirect you to the editor as normal.
